requirements.txt

```diff
@@ -7,3 +7,4 @@ toml
 appdirs
 semantic_version==2.6
 dask[delayed]
+autoray
```

Comment:
Should we lock a version?

Reply:
by convention, we really should be pinning in the `requirements.txt`. The only reason we aren't is a bit of a historical accident - the CI installs the environment from this file, and it helps us catch bugs/issues when there are dependency upgrades 🤔 

What we should do, really, is have a separate `requirements.txt` (pinned) and `requirements-ci.txt` (unpinned).